### DISCUSSION TITLE
Pattern for I18N

### DISCUSSION THREAD

**Discussion by jkies00** [smarty-php/smarty#912] [General]
Hi All,

Is there an existing pattern for internationalization for Smarty?

In platforms I've worked with in the past, there were language packs that would be applied based on user locale, or user language selection.

Is there something like this for Smarty?

Thanks

**Comment by saulery** [comment]
Hello,

Check [smarty-gettext](https://github.com/smarty-gettext/smarty-gettext). I use it to translate the interface of a B2B site into 5 languages. I added one or two personal improvements but it's solid.

Regards,

**Comment by juangacovas** [comment]
+1 for that package, my team uses it too for i18n with smarty
